### PR TITLE
Added blocks for Dynamic Content

### PR DESCRIPTION
**Pull request by bulicamihai**
https://github.com/Financial-Times/upp-ftcom-lovin/issues/44

### CODE REVIEW

**Comment by bulicamihai** [comment]
[Process DC blocks in MAICM](https://trello.com/c/VyojPDeb/159-process-dc-blocks-in-maicm)

File: src/main/java/com/ft/methodearticleinternalcomponentsmapper/exception/MethodeArticleMarkedDeletedException.java [hunk 1]
```diff
@@ -4,7 +4,13 @@
 
 public class MethodeArticleMarkedDeletedException extends RuntimeException {
```
Comment (MihaiSorinMoisa) [inline src/main/java/com/ft/methodearticleinternalcomponentsmapper/exception/MethodeArticleMarkedDeletedException.java]: Make sure this exception is consistent with the one from MAM.

File: src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java [hunk 2]
```diff
@@ -6,25 +6,15 @@
...
-import com.ft.methodearticleinternalcomponentsmapper.model.AlternativeStandfirsts;
-import com.ft.methodearticleinternalcomponentsmapper.model.AlternativeTitles;
-import com.ft.methodearticleinternalcomponentsmapper.model.Design;
-import com.ft.methodearticleinternalcomponentsmapper.model.EomFile;
-import com.ft.methodearticleinternalcomponentsmapper.model.Image;
-import com.ft.methodearticleinternalcomponentsmapper.model.InternalComponents;
-import com.ft.methodearticleinternalcomponentsmapper.model.Summary;
-import com.ft.methodearticleinternalcomponentsmapper.model.TableOfContents;
-import com.ft.methodearticleinternalcomponentsmapper.model.Topper;
+import com.ft.methodearticleinternalcomponentsmapper.model.*;
```
Comment (MihaiSorinMoisa) [inline src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java]: Don't use wildcard imports.

File: src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java [hunk 3]
```diff
@@ -6,25 +6,15 @@
...
 import com.ft.methodearticleinternalcomponentsmapper.validation.PublishingStatus;
 import com.ft.uuidutils.DeriveUUID;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Maps;
 import org.apache.commons.lang.StringUtils;
-import org.w3c.dom.Document;
-import org.w3c.dom.Element;
-import org.w3c.dom.Node;
+import org.w3c.dom.*;
```
Comment (MihaiSorinMoisa) [inline src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java]: Don't use wildcard imports.

File: src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java [hunk 4]
```diff
@@ -235,12 +234,16 @@ private String transformBody(XPath xpath, String sourceBodyXML, Document attribu
         return postProcessedTransformedBody;
     }
 
-    private String determineType(final XPath xpath, final Document attributesDocument) throws XPathExpressionException, TransformerException {
+    private String determineType(final XPath xpath, final Document attributesDocument, String sourceCode) throws XPathExpressionException, TransformerException {
```
Comment (MihaiSorinMoisa) [inline src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java]: Transformer exception is never thrown from this method.

File: src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java [hunk 5]
```diff
@@ -121,25 +116,26 @@ public InternalComponentsMapper(FieldTransformer bodyTransformer,
     public InternalComponents map(EomFile eomFile, String transactionId, Date lastModified, boolean preview) {
         try {
             UUID uuid = UUID.fromString(eomFile.getUuid());
+            final XPath xpath = XPathFactory.newInstance().newXPath();
+            final Document attributesDocument = getAttributesDocument(eomFile);
+            final Document valueDocument = getValueDocument(eomFile);
 
             String sourceCode = retrieveSourceCode(eomFile.getAttributes());
```
Comment (MihaiSorinMoisa) [inline src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java]: Replace `retrieveSourceCode(eomFile.getAttributes())` with `xpath.evaluate(SOURCE_ATTR_XPATH, attributesDocument)`.

File: src/main/java/com/ft/methodearticleinternalcomponentsmapper/model/InternalComponents.java [hunk 6]
```diff
@@ -20,11 +20,14 @@
     private final AlternativeTitles alternativeTitles;
     private final AlternativeStandfirsts alternativeStandfirsts;
     private final String pushNotificationsCohort;
+    private final String type;
```
Comment (MihaiSorinMoisa) [inline src/main/java/com/ft/methodearticleinternalcomponentsmapper/model/InternalComponents.java]: Why do we need the `type` field for internalcomponents?

**Pull request closed without merging**